File: collectors/openbus.hypervisor/src/main/java/br/com/produban/openbus/hypervisor/config/DevicesConfiguration.java
package br.com.produban.openbus.hypervisor.config;

import java.io.File;
import java.util.List;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.util.StringUtils;

import com.typesafe.config.ConfigFactory;

@Configuration
public class DevicesConfiguration {

	@Value("${config.file.path}")
	private String configFilePath;

	@Bean
	public List<String> ipAddresses() {

		if (StringUtils.isEmpty(configFilePath)) {
			return ConfigFactory.load("devices").getStringList("devices.ipAddresses");
		}

		return ConfigFactory.parseFile(new File(configFilePath)).getStringList("devices.ipAddresses");
	}
}

File: collectors/openbus.f5box/src/main/scala/br/com/produban/openbus/f5/configuration/DevicesConfiguration.scala
package br.com.produban.openbus.f5.configuration

import java.io.File

import com.typesafe.config.ConfigFactory
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.{Bean, Configuration}

import scala.collection.JavaConverters._

@Configuration
class DevicesConfiguration {

  val LOGGER = LoggerFactory.getLogger(classOf[DevicesConfiguration])

  @Value("${f5.devices.config.path}")
  var devicesConfigPath: String = _

  @Bean
  def ipAddresses(): List[String] = {
    val configFromPath = ConfigFactory.parseFile(new File(devicesConfigPath))

    if (configFromPath.isEmpty) {
      LOGGER.info("External config not specified ... Loading config from classpath !")
      ConfigFactory.load("devices").getStringList("devices.ipAddresses").asScala.toList
    } else {
      LOGGER.info("Loading external configuration from=" + devicesConfigPath)
      configFromPath.getStringList("devices.ipAddresses").asScala.toList
    }
  }
}
